### DISCUSSION TITLE
SRDF Formulation

### DISCUSSION THREAD

**Discussion by fishbotics** [tesseract-robotics/tesseract#800] [General]
Hi all,

Is there any documentation on the collision matrix SRDF formulation? I used `tesseract-ignition` to generate an SRDF file using two nearly identical URDFs and the collision matrices generated were very similar but slightly different and I'd like to understand why, as well as what I can manually change. 

One example is that the robot has two fingers that are each connected to the hand. These fingers are not technically adjacent, but they mimic each other (it's specified in the URDF) and, because they are for grabbing, they should sometimes be able to collide. 

With one URDF, `tesseract-ignition` generated that these collision should be ignored with the reason `Never`, but the other URDF did not generate the same element. I'd like to copy that line over between the two `srdf` files. Is that okay? And, does the reason for the entry in the collision matrix matter? It seems like the choices are `Allways` `Never` and `Adjacent`

Thanks!

**Comment by Levi-Armstrong** [comment]
There is no issue with coping them over or adding them manually. The reason can be anything and is used just to describe why it was disable and the auto generation is only able to detect the three. There can be difference between two sets of URDF because the resolution at which the URDF is sampled depends on the sum of all of the joint ranges. I the majority of case the auto generated allowed collision matrix will be good, but it should be reviewed for correctness.
